src/librustc_resolve/diagnostics.rs

```diff
@@ -948,6 +948,50 @@ use something_which_doesnt_exist;
 Please verify you didn't misspell the import's name.
 "##,
 
+E0434: r##"
+This error indicates that a variable usage inside an inner function is invalid
+because the variable comes from a dynamic environment. Inner functions do not
+have access to their containing environment.
```

Comment:
Either you put "Example of erroneous code:" directly after the text, or you separate it from the first block (and then add an empty line between the two to make the separation obvious).